crates/testing/tests/tests_1/quorum_proposal_task.rs

```diff
@@ -576,7 +592,9 @@ async fn test_quorum_proposal_task_liveness_check() {
             SendPayloadCommitmentAndMetadata(
                 build_payload_commitment::<TestTypes>(&quorum_membership, ViewNumber::new(1)),
                 builder_commitment.clone(),
-                TestMetadata,
+                TestMetadata {
```

Comment:
one concern i have here is a lot of places we set num_transactions to 0. if we add another field to the `TestMetadata` down the line and it also has some default value we will have to change all these files again. I dont know if its worth implementing default and changing everywhere we set to 0.

Reply:
I'm a little wary of `default` here because I don't think there's necessarily a "default" for what the metadata should be, especially in this case since it's actually tied to something else.

But I do think these tests should really try to take their fields from a block header (since everything is coming from a view generator we should have access to one for each view) instead of manually constructing it; let me see if I can fix that